Adding continuous tabulating function

ddsjoberg opened the pull request:
To do...
- [ ] What should the final name of the function be?
- [ ] Is `col=` the best name for the argument? Maybe `by=` is better to be more like `tbl_summary()` (instead of being more like `tbl_cross()`)? I think I prefer `by=` and to change the function name to not be similar to `tbl_cross()`?
- [ ] When there is a `col=` variable passed, the underlying column names should probably be `stat_1`, `stat_2` like other similar tables. 
- [ ] Do I need to do anything to be able to use the "`{level}`" functionality in `modify_header()`? Maybe need to return `df_by` and `N` in the return list?

**What changes are proposed in this pull request?**


**If there is an GitHub issue associated with this pull request, please provide link.**


--------------------------------------------------------------------------------

Checklist for PR reviewer

- [ ] PR branch has pulled the most recent updates from master branch. Ensure the pull request branch and your local version match and both have the latest updates from the master branch.
- [ ] If an update was made to `tbl_summary()`, was the same change implemented for `tbl_svysummary()`?
- [ ] If a new function was added, function included in `_pkgdown.yml`
- [ ] If a bug was fixed, a unit test was added for the bug check
- [ ] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [ ] Code coverage is suitable for any new functions/features. Review coverage with `covr::report()`. Before you run, set `Sys.setenv(NOT_CRAN="true")` and begin in a fresh R session without any packages loaded. 
- [ ] R CMD Check runs without errors, warnings, and notes
- [ ] `usethis::use_spell_check()` runs with no spelling errors in documentation

When the branch is ready to be merged into master:
- [ ] Update NEWS.md with the changes from this pull request under the heading "`# gtsummary (development version)`". If there is an issue associated with the pull request, reference it in parantheses at the end update (see `NEWS.md` for examples).
- [ ] Increment the version number using `usethis::use_version(which = "dev")` 
- [ ] Run `codemetar::write_codemeta()`
- [ ] Run `usethis::use_spell_check()` again
- [ ] Approve Pull Request
- [ ] Merge the PR. Please use "Squash and merge".

ddsjoberg commented:
Decisions made:
1. `tbl_continuous()` 
2. Make args more like `tbl_summary()`, e.g. `tbl_continuous(con=,  include=, by=)`

larmarange commented:
For your information, it is also very easy to reproduce the behavior of `tbl_continuous()`.

``` r
library(gtsummary)
#> #BlackLivesMatter
continuous_summary <- function(variable_to_summarize) {
  function(data, stat_display, ...) {
    gtsummary:::summarize_continuous(
      data,
      variable = variable_to_summarize,
      by = NULL, 
      stat_display = stat_display,
      summary_type = "continuous"
    ) %>%
      dplyr::select(-.data$variable, -.data$stat_display)
  }
}


trial %>% 
  dplyr::group_by(trt, grade) %>%
  dplyr::group_modify(
    continuous_summary("age"),
    stat_display = "{median} ({p25}, {p75})"
  )
#> # A tibble: 6 x 5
#> # Groups:   trt, grade [6]
#>   trt    grade median   p25   p75
#>   <chr>  <fct>  <dbl> <dbl> <dbl>
#> 1 Drug A I       46    36    60  
#> 2 Drug A II      44.5  31    54.5
#> 3 Drug A III     51.5  42.2  60.2
#> 4 Drug B I       48    42    55  
#> 5 Drug B II      50.5  43    57.2
#> 6 Drug B III     45    36    52
trial %>%
  tbl_custom_summary(
    include = grade,
    by = trt,
    stat_fns = everything() ~ continuous_summary("age"),
    statistic = everything() ~ "{median} ({p25}, {p75})"
  ) %>%
  add_overall() %>%
  as_kable()
```

| **Characteristic** | **Overall**, N = 200 | **Drug A**, N = 98 | **Drug B**, N = 102 |
|:-------------------|:---------------------|:-------------------|:--------------------|
| Grade              |                      |                    |                     |
| I                  | 47 (37, 56)          | 46 (36, 60)        | 48 (42, 55)         |
| II                 | 48 (37, 57)          | 44 (31, 54)        | 50 (43, 57)         |
| III                | 47 (38, 58)          | 52 (42, 60)        | 45 (36, 52)         |

<sup>Created on 2021-09-11 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

karissawhiting commented:
@ddsjoberg This feature is really useful. I'm looking forward to using it!

An initial thought about the `statistic` argument: 

I was thinking it would be useful to accept a list of formulas and allow different continuous statistic types for different comparison variables. 
 
E.g. 

```
tbl_continuous(
  data = trial,
  variable = age,
  statistic =  list(grade ~ "{mean}", 
                    trt ~ "{median}"),
  include = c(trt, grade)
)
```

However, using it this way is a llittle inconsistent with `left side variable ~ right side statistic` relationship in `tbl_summary()`, so that could potentially be confusing?

Conversely, you could acheive the same result this way:

```
x <- tbl_continuous(
  data = trial,
  variable = age,
  statistic =  "{mean}",
  include = c(grade)
)

y <- tbl_continuous(
  data = trial,
  variable = age,
  statistic =  "{median}",
  include = c(trt)
)

tbl_stack(list(x, y))
```

But it's a bit more cumbersome and the footnote needs adjustment. 

Thoughts on this?

ddsjoberg commented:
Thank you for the review @karissawhiting !!

You make a very good point, and this is something I planned very poorly unfortunately. The `tbl_summary(statistic=)` argument take a formula list, `tbl_cross(statistic=)` is a string, and it's not consistent. UGH

Since `tbl_summary()` is more often used, matching that syntax is preferable I think. I"ll see how difficult that update is, and report back!

ddsjoberg commented:
FYI @larmarange @karissawhiting I am going to update this function to use Joseph's contribution `tbl_custom_summary()`: it'll make the internals sooo much nicer to read :)

ddsjoberg commented:
Hey @karissawhiting @larmarange !

I updated the back-end to use `tbl_custom_summary()`.....so much less code! Very nice! Thanks @larmarange !

this allows users to select different stats for variable variables and I also added a digits argument to control rounding of the continuous variable.

larmarange commented:
That's great @ddsjoberg

- Would it be relevant to add the option `overall_row`? Could be useful here.
- Just curious, why duplicating `continuous_summary ()`
